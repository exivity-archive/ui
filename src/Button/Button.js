import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 ${props => props.theme.spacing}rem;
  padding:
    ${props => props.theme.spacing / 2}rem
    ${props => props.theme.spacing * 2}rem;

  ${props =>
    props.primary && css`
      background: ${props.theme.colours.primary};
    `}
`

const StyledButton = ({ children, primary }) => <Button primary={primary}>
  {children}
</Button>

StyledButton.propTypes = {
  children: PropTypes.node.isRequired,
  primary: PropTypes.bool
}

export default StyledButton
