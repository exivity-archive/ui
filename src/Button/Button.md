Default Button:

```js
<Button>Submit</Button>
```

More pink is more better:

```js
<Button pink>ğŸ¦„</Button>
```
