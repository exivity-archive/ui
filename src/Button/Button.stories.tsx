import React from 'react'
import styled from 'styled-components'
import { storiesOf } from '@storybook/react'
import { MdClear, MdCloud, MdFavorite, MdSave, MdWhatshot } from 'react-icons/md'

import { Icon } from '../Icon'
import { Button } from '.'
import { Row } from '../utils/stories/components'
import { fromTheme } from '../utils/styled'

const Container = styled.div`
  display: flex;
  background-color: ${fromTheme(theme => theme.colors.lightGray)}
  padding: 20px;
`

storiesOf('interact|Button', module)
  .add('default', () => <Button>Default</Button>)
  .add('purposes', () => <Row>
    <Button primary>Primary</Button>
    <Button secondary>Secondary</Button>
    <Button success>Success</Button>
    <Button danger>Danger</Button>
  </Row>)
  .add('sizes', () => <Row>
    <Button tiny>Tiny</Button>
    <Button small>Small</Button>
    <Button>Default</Button>
    <Button large>Large</Button>
  </Row>)
  .add('outlined', () => <Row>
    <Button outlined primary>Primary</Button>
    <Button outlined secondary>Secondary</Button>
    <Button outlined success>Success</Button>
    <Button outlined danger>Danger</Button>
  </Row>)
  .add('icon', () => <Button><Icon><MdFavorite /></Icon> Favorite</Button>)
  .add('disabled', () => <Button disabled>Disabled</Button>)
  .add('round', () => <Row columns={4}>
    <Button round primary>☁️</Button>
    <Button round secondary>🔥</Button>
    <Button round success>💾</Button>
    <Button round danger>❌</Button>
    <Button round outlined primary>☁️</Button>
    <Button round outlined secondary>🔥</Button>
    <Button round outlined success>💾</Button>
    <Button round outlined danger>❌</Button>
    <Button round tiny>❤️️</Button>
    <Button round small>❤️️</Button>
    <Button round>❤️️</Button>
    <Button round large>❤️️</Button>
    <Button round outlined tiny>❤️️</Button>
    <Button round outlined small>❤️️</Button>
    <Button round outlined>❤️️</Button>
    <Button round outlined large>❤️️</Button>
    <Button round primary><Icon><MdCloud /></Icon>️</Button>
    <Button round secondary><Icon><MdWhatshot /></Icon></Button>
    <Button round success><Icon><MdSave /></Icon></Button>
    <Button round danger><Icon><MdClear /></Icon></Button>
    <Button round outlined primary><Icon><MdCloud /></Icon>️</Button>
    <Button round outlined secondary><Icon><MdWhatshot /></Icon></Button>
    <Button round outlined success><Icon><MdSave /></Icon></Button>
    <Button round outlined danger><Icon><MdClear /></Icon></Button>
    <Button round tiny><Icon><MdFavorite /></Icon>️</Button>
    <Button round small><Icon><MdFavorite /></Icon>️</Button>
    <Button round><Icon><MdFavorite /></Icon>️</Button>
    <Button round large><Icon><MdFavorite /></Icon>️</Button>
    <Button round outlined tiny><Icon><MdFavorite /></Icon>️</Button>
    <Button round outlined small><Icon><MdFavorite /></Icon>️</Button>
    <Button round outlined><Icon><MdFavorite /></Icon>️</Button>
    <Button round outlined large><Icon><MdFavorite /></Icon>️</Button>
    <span/>
  </Row>)
  .add('transparent', () => (
    <Container>
      <Row columns={8}>
        <Button round outlined tiny transparent><Icon><MdFavorite /></Icon></Button>
        <Button round small transparent><Icon><MdFavorite /></Icon></Button>
        <Button round transparent><Icon><MdFavorite /></Icon></Button>
        <Button round large transparent><Icon><MdFavorite /></Icon></Button>
        <Button outlined primary transparent>Primary</Button>
        <Button outlined secondary transparent>Secondary</Button>
        <Button outlined success transparent>Success</Button>
        <Button outlined danger transparent>Danger</Button>
      </Row>
    </Container>
  ))
