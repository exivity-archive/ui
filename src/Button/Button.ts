import styled from 'styled-components'

const Button = styled.button`
  padding: 20px;
  background-color: hotpink;
`

Button.defaultProps = {}

/** @component */
export default Button
