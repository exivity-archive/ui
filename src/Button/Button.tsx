import React from 'react'
import styled, { css } from 'styled-components'
import { InputProps } from '../AbstractInput/AbstractInput'
import { Icon } from '../Icon'
import { fromTheme, hexToString, matchThemeProp } from '../utils/styled'
import { preciseEm } from '../utils/styled/isolated'

export interface ButtonProps extends InputProps {
  // Variants
  round?: boolean
}

export const Button = styled.button<ButtonProps>`
  font-family: ${fromTheme(theme => theme.global.fontFamily)};
  font-weight: 500;
  font-size: ${matchThemeProp(theme => theme.global.sizes, {
    modifier: (em: number) => em / 16 * 14
  })}em;
  color: ${fromTheme(theme => theme.colors.white)};
  line-height: ${fromTheme(theme => theme.global.lineHeight)};

  text-transform: uppercase;
  display: inline-flex;
  position: relative;
  appearance: none;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: ${props => props.tiny
    ? `${preciseEm(2.5, 10)}em` // 25px
    : `${preciseEm(2.85, 14)}em`}; // 40px = 2.5 * 16
  padding: 0 ${preciseEm(1.5)}em;
  border: none;
  border-radius: ${fromTheme(theme => theme.global.borderRadius)}px;
  flex: none;
  user-select: none;
  white-space: nowrap;
  text-decoration: none;
  outline: none;

  background-color: ${matchThemeProp(theme => theme.global.purposes)};
  --focus-color: ${matchThemeProp(theme => theme.global.purposes, { modifier: hexToString })};

  &:hover {
    box-shadow: inset 10px 10px 999em rgba(0,0,0,${fromTheme(theme => theme.global.shadowAlpha)});
  }

  &:focus {
    box-shadow: inset 0 0 999em rgba(0,0,0,${fromTheme(theme => theme.global.shadowAlpha)}),
      0 0 0 ${fromTheme(theme => theme.global.outlineWidth)}px rgba(var(--focus-color), ${fromTheme(theme => theme.global.outlineAlpha)});
  }

  &:active,
  &.active {
    box-shadow: inset 0 0 999em rgba(0,0,0,${fromTheme(theme => theme.global.shadowAlpha * 2)}),
      0 0 0 ${fromTheme(theme => theme.global.outlineWidth)}px rgba(var(--focus-color), ${fromTheme(theme => theme.global.outlineAlpha)});
  }

  &:after {
    display: none;
    content: "";
    position: absolute;
    top: -${fromTheme(theme => theme.global.borderWidth)}px;
    right: -${fromTheme(theme => theme.global.borderWidth)}px;
    bottom: -${fromTheme(theme => theme.global.borderWidth)}px;
    left: -${fromTheme(theme => theme.global.borderWidth)}px;
    border-radius: inherit;
    background-color: rgba(255, 255, 255, 0.5);
  }

  &[disabled] {
    pointer-events: none;
    &:after {
      display: block;
    }
  }

  ${(props: ButtonProps) => props.outlined && css`
    background-color: ${fromTheme(theme => theme.colors.white)};
    color: ${matchThemeProp(theme => theme.global.purposes)};
    box-shadow: 0 0 0 ${fromTheme(theme => theme.global.borderWidth)}px ${matchThemeProp(theme => theme.global.purposes)};

    &:hover {
      color: ${fromTheme(theme => theme.global.textColor)};
      box-shadow: 0 0 0 ${fromTheme(theme => theme.global.borderWidth)}px ${fromTheme(theme => theme.global.textColor)};
    }

    &:focus {
      box-shadow: 0 0 0 ${fromTheme(theme => theme.global.borderWidth)}px ${matchThemeProp(theme => theme.global.purposes)},
       0 0 0 ${fromTheme(theme => theme.global.outlineWidth + 1)}px rgba(var(--focus-color), ${fromTheme(theme => theme.global.outlineAlpha)});
    }
  `}

  ${(props: ButtonProps) => props.round && css`
    border-radius: 50%;
    min-width: auto;
    padding: 0;
    width: ${props.tiny
    ? preciseEm(2.5, 10) // 25px
    : preciseEm(2.85, 14)}em; // 40px = 2.5 * 16
  `}

  ${Icon} {
    font-size: 1.5em;
    ${(props: ButtonProps) => props.round !== true && css`
      margin-right: ${fromTheme(theme => theme.global.baseSpacing / 2)}em;
    `}
  }
`
