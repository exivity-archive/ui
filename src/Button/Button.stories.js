import React from 'react'
import { Grid } from 'reakit'

import { storiesOf } from '@storybook/react'

import Button from './Button'
import Icon from './../Icon'

storiesOf('atoms|Button', module)
  .add('default', () => <Button>Button</Button>)
  .add('palette', () => <Grid gridTemplateColumns='repeat(5, max-content)' gap={20} align>
    {[0, 1, 2, 3, 4].map(tone => <React.Fragment>
      <Button primary tone={tone}>Primary</Button>
      <Button secondary tone={tone}>Secondary</Button>
      <Button success tone={tone}>Success</Button>
      <Button warning tone={tone}>Warning</Button>
      <Button danger tone={tone}>Danger</Button>
    </React.Fragment>)}
  </Grid>)
  .add('size', () => <Grid gridTemplateColumns='repeat(5, max-content)' gap={20} align>
    <Button primary xsmall>Extra small</Button>
    <Button primary small>Small</Button>
    <Button primary>Normal</Button>
    <Button primary large>Large</Button>
    <Button primary xlarge>Extra large</Button>
  </Grid>)
  .add('icon', () => <Grid gridTemplateColumns='repeat(2, max-content)' gap={20} align>
    <Button primary xlarge><Icon>favorite</Icon> like</Button>
    <Button success xlarge rightIcon>retweet <Icon style={{ marginRight: 0 }}>sync</Icon></Button>
  </Grid>)
