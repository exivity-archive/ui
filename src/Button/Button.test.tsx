import React from 'react'
import renderer from 'react-test-renderer'

import Button from './Button'

test('renders basic button', () => {
  const button = renderer.create(<Button />)
  expect(button.toJSON()).toMatchSnapshot()
})
