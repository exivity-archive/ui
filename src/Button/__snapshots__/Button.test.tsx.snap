// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic button 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #00a8d8;
  --focus-color: 0,168,216;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different purpose buttons 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #00a8d8;
  --focus-color: 0,168,216;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different purpose buttons 2`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #999;
  --focus-color: 153,153,153;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different purpose buttons 3`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #4caf50;
  --focus-color: 76,175,80;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different purpose buttons 4`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #ef5350;
  --focus-color: 239,83,80;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different size buttons 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 16.625px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #00a8d8;
  --focus-color: 0,168,216;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders different size buttons 2`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 12.25px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #00a8d8;
  --focus-color: 0,168,216;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;

exports[`renders outlined button 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: 500;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-transform: uppercase;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  min-width: 2.5em;
  height: 2.857142857142857em;
  padding: 0 1.5em;
  border: none;
  border-radius: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  background-color: #00a8d8;
  --focus-color: 0,168,216;
  background-color: #fff;
  color: #00a8d8;
  box-shadow: 0 0 0 1px #00a8d8;
}

.c0:hover {
  box-shadow: inset 10px 10px 999em rgba(0,0,0,0.1);
}

.c0:focus {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.1), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:active,
.c0.active {
  box-shadow: inset 0 0 999em rgba(0,0,0,0.2), 0 0 0 4px rgba(var(--focus-color),0.15);
}

.c0:after {
  display: none;
  content: "";
  position: absolute;
  top: -1px;
  right: -1px;
  bottom: -1px;
  left: -1px;
  border-radius: inherit;
  background-color: rgba(255,255,255,0.5);
}

.c0[disabled] {
  pointer-events: none;
}

.c0[disabled]:after {
  display: block;
}

.c0:hover {
  color: #222;
  box-shadow: 0 0 0 1px #222;
}

.c0:focus {
  box-shadow: 0 0 0 1px #00a8d8, 0 0 0 5px rgba(var(--focus-color),0.15);
}

.c0 {
  margin-right: 0.625em;
}

<button
  className="c0"
/>
`;
